### PR TITLE
Lift comment state into discussion

### PR DESCRIPTION
**Pull request by abeddow91**
## What does this change?
Lifts up comment state into the discussion component and out of comments. This means we only make one API call, and don't duplicate state. 

It also removes the useDiscussion hook and implements the useCommentCount hook in discussion and discussion meta.

It also updates the comment stories to use manual fixture data. 

## Why?
We want discussion to have the source of truth for shared state. This makes the discussion app easier to read and reason about. 

[This is part of ](https://github.com/guardian/dotcom-rendering/issues/10228)

### CODE REVIEW

File: dotcom-rendering/src/components/Discussion.tsx [hunk 1]
```diff
@@ -91,22 +92,38 @@ export const Discussion = ({
 	user,
 	idApiUrl,
 }: Props) => {
-	const [commentPage, setCommentPage] = useState<number>(1);
```
Comment (abeddow91) [inline dotcom-rendering/src/components/Discussion.tsx:94]: remove types where they can be inferred

File: dotcom-rendering/src/components/Discussion.tsx [hunk 2]
```diff
@@ -91,22 +92,41 @@ export const Discussion = ({
...
+	const [commentPage, setCommentPage] = useState(1);
 	const [commentPageSize, setCommentPageSize] = useState<25 | 50 | 100>();
 	const [commentOrderBy, setCommentOrderBy] = useState<
 		'newest' | 'oldest' | 'recommendations'
 	>();
-	const [isExpanded, setIsExpanded] = useState<boolean>(false);
+	const [commentCount, setCommentCount] = useState(0);
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Discussion.tsx:100]: We have `useCommentCount` for this specific purpose

File: dotcom-rendering/src/components/Discussion.tsx [hunk 3]
```diff
@@ -172,13 +192,6 @@ export const Discussion = ({
 		}
 	}, []);
 
-	useEffect(() => {
-		// There's no point showing the view more button if there isn't much more to view
-		if (commentCount === 0 || commentCount === 1 || commentCount === 2) {
-			setIsExpanded(true);
-		}
-	}, [commentCount]);
-
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Discussion.tsx:181]: I think that the pattern of having state being derived from other state is pretty sound. Why fold this into `getDiscussion`’s then?

File: dotcom-rendering/src/components/Discussion.tsx (hunk 2 above)
Comment (abeddow91) [reply dotcom-rendering/src/components/Discussion.tsx:100]: Ah, great, thanks Max! I didn't know that existed. I'll update the code to use that 👍

**Review by mxdvl — COMMENTED** [review]
This effectively makes `useDiscussion` only used in a single location. What are the plans for that custom hook?

File: dotcom-rendering/src/components/Discussion.tsx (hunk 3 above)
Comment (mxdvl) [reply dotcom-rendering/src/components/Discussion.tsx:181]: Of course, that makes sense, thanks for clarifying!

**Comment by abeddow91** [comment]
>This effectively makes useDiscussion only used in a single location. What are the plans for that custom hook? 

Its a good question! I'm unsure how useful it is as a hook really. It returns `commentCount` and `isClosedForComments` only and given that we now have the useCommentCount hook, its probably only useful for isClosedForComments.

**Review by mxdvl — APPROVED** [review]
Unify all the states!

File: dotcom-rendering/src/components/Discussion.tsx (hunk 2 above)
Comment (mxdvl) [reply dotcom-rendering/src/components/Discussion.tsx]: I’ve actually discovered that `commentCount` and `topLevelCommentCount` cannot be conflated… the former relates to the number of comments, the latter to the number of threads!

Fix in #10477

**Pull request merged**